dapps/tezos-provider/src/utils/tezos-provider.ts

```diff
@@ -0,0 +1,380 @@
+      .then((data) => {
+        return data
+          .map((op: any) => {
+            const address = op?.status === 'applied' && op?.originatedContract?.kind === "smart_contract" ? op.originatedContract.address : '';
+            if (address) {
+              console.log('Got contract address:', address);
+            }
+            return address;
+          })
+          .filter((address: string) => address.length);
+      });
+  }
+
+  public async checkConnection(): Promise<boolean> {
+    if (!this.isConnected) {
+      throw new TezosConnectionError();
+    }
+    return true;
+  }
+  
+  // Requests using the WalletConnect connection
+  
+  public async tezosGetAccounts(): Promise<TezosGetAccountResponse> {
+    if (!this.signer) {
+      throw new TezosInitializationError();
+    }
+    await this.checkConnection();
+  
+    const result = await this.signer.request<TezosGetAccountResponse>({
+      method: DEFAULT_TEZOS_METHODS.TEZOS_GET_ACCOUNTS,
+      params: {},
+    }, this.chainId);
+    this.accounts = result.map((account) => account.address);
+  
+    return result;
+  }
+
+  // Method to sign a message
+  public async tezosSign(payload: string): Promise<TezosSignResponse> {
+    if (!this.signer) {
+      throw new TezosInitializationError();
+    }
+    await this.checkConnection();
+    
+    const result = await this.signer.request<TezosSignResponse>({
+      method: DEFAULT_TEZOS_METHODS.TEZOS_SIGN,
+      params: {
+        account: this.address,
+        payload,
+      },
+    }, this.chainId);
+
+    return result;
+  }
+
+  // Method to send operations
+  public async tezosSend(op: PartialTezosOperation): Promise<TezosSendResponse> {
+    if (!this.signer) {
+      throw new TezosInitializationError();
+    }
+    await this.checkConnection();
+    
+    const result = await this.signer.request<TezosSendResponse>({
+      method: DEFAULT_TEZOS_METHODS.TEZOS_SEND,
+      params: {
+        account: this.address,
+        operations: [op],
+      },
+    }, this.chainId);
+
+    return result;
+  }
+
+  // Method to send a transaction
+  public async tezosSendTransaction(op: PartialTezosTransactionOperation): Promise<TezosSendResponse> {
+    return this.tezosSend(op);
+  }
+
+  // Method to send a delegation
+  public async tezosSendDelegation(op: PartialTezosDelegationOperation): Promise<TezosSendResponse> {
+    return this.tezosSend(op);
+  }
+
+  // Method to send an undelegation
+  public async tezosSendUndelegation(): Promise<TezosSendResponse> {
+    const op: PartialTezosDelegationOperation = { kind: TezosOperationType.DELEGATION };
+    return this.tezosSend(op);
+  }
+
+  // Method to originate a contract
+  public async tezosSendOrigination(op: PartialTezosOriginationOperation): Promise<TezosSendResponse> {
+    return this.tezosSend(op);
+  }
+
+  // Method to call a smart contract
+  public async tezosSendContractCall(op: PartialTezosTransactionOperation): Promise<TezosSendResponse> {
```

Comment:
Are we planning to include all the supported operations here in the provider ? or only these 5?

Reply:
We can add other operations (I suggest to do this in the next PR). Here is the full list (TZIP-10):
```
export type PartialTezosOperation =
  | TezosActivateAccountOperation
  | TezosBallotOperation
  | PartialTezosDelegationOperation
  | TezosDoubleBakingEvidenceOperation
  | TezosEndorsementOperation
  | PartialTezosOriginationOperation
  | TezosProposalOperation
  | PartialTezosRevealOperation
  | TezosSeedNonceRevelationOperation
  | PartialTezosTransactionOperation;
```
Covered:
 - `PartialTezosDelegationOperation`
 - `PartialTezosOriginationOperation`
 - `PartialTezosTransactionOperation`
Not covered yet:
 - `TezosActivateAccountOperation`
 - `TezosBallotOperation`
 - `TezosDoubleBakingEvidenceOperation`
 - `TezosEndorsementOperation`
 - `TezosProposalOperation`
 - `PartialTezosRevealOperation`
 - `TezosSeedNonceRevelationOperation`
 The problem with them is that I don't know how to make valid requests.
 I can easily add support for them and (if desired) I can add them to the example, but they will fail since the easiest way is to fill unavailable data with dummy data.
 What do you think?